src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -97,8 +98,24 @@ private void getMealList(HttpServletRequest request, HttpServletResponse respons
         // Every object Meal adds if it contains at least one keyword in at least one field 
         // (type, title, description, ingredients).
         List<Meal> searchedMeal = new ArrayList<>();
+        HashMap<Integer, List<Meal>> sortedMeal = new HashMap<>();
         for (Meal meal : meals) {
-            if (isResultOfSearch(meal, params)) {
+            int count = isResultOfSearch(meal, params);
+            if (count > 0) {
+                List<Meal> list = new ArrayList<>();
+                if (sortedMeal.get(count) != null) {
+                    list = sortedMeal.get(count);
+                }
+                list.add(meal);
+                sortedMeal.put(count, list);
+            }
+        }
+        for (int i = params.size(); i > 0; i --) {
+            if (sortedMeal.get(i) == null) {
+                continue;
+            }
+            List<Meal> list = sortedMeal.get(i);
```

Comment:
What about 
```
List<Meal> list = sortedMeal.get(i);
if (list != null) {
  for (Meal meal : list) {
    searchedMeal.add(meal);
  }
}

```

not to call `sortedMeal.get(i)` twice and make code a little bit more readable